### ISSUE TITLE
[QUESTION] Consumer sees no messages

### ISSUE THREAD

**Issue by Putr** [mateusjunges/laravel-kafka#183]
Hello,

Using this code to consume a topic, that, using Kowl and kcat I can see has hundreds of messages in >20 partitions:

```php
$consumer = Kafka::createConsumer(
	['topic'],
	'group-name',
	'server:9094'
)
  ->withHandler(
    function (KafkaConsumerMessage $message) {
	    dd($message);
    }
  )
  ->stopAfterLastMessage()
  ->withMaxMessages(1)
  ->build()
  ->consume();
```

The consumer prints no message (even tried writing to Log - still nothing) and exits (or just sits there without stopAfterLastmessage()).

This is my first time using Kafka, so what am I missing? Why does kcat read the messages, but my code does not?

**Comment by drmnk** [comment]
Same here, when I don't chain _->stopAfterLastMessage()_ my consumer reads new messages and continues it's work (undesirable behavior, cause I want to start it by schedule).

But if I use _->stopAfterLastMessage()_ I just get null and messages doesn't handle

**Comment by drmnk** [comment]
I think the deal is that _->stopAfterLastMessage()_ stops message handling not only when it's real last message, but also when the $message->err code is not 0 (Consumers\Consumer\handleMessage)

Because of something some of my Kafka messages gets error code -185

**Comment by Putr** [comment]
I get no messages – at all. My handler never gets triggered. But I know there are messages waiting.

**Comment by drmnk** [comment]
Neither mine handlers, so I put var_dump in Consumer.php to see that I get
error on the very first message.

Anyway, I solve the issue by using Rdkafka directly with same
configuration.

Later I will try to provide two pieces of code when Rdkafka works and
Laravel Kafka is not

Пт, 24 марта 2023 г. в 18:17, Rok Andrée ***@***.***>:

**Comment by Putr** [comment]
For anyone finding this later:

The issues was, that kafka tracks "read" status of messages based on "index" of the message tied to a group - so in group X all messages up to index Z have been read.

For some reason this library never showed any messages, even when I changed the group but acted like they are all read. Even with autocommit disabled. Never figured out why. 

I'm now using the underlying library directly and have had no issues.

**Comment by mateusjunges** [comment]
Hey folks, I'm late to the party but I discoverd a bug in the `auto-commit` feature. The `DefaultCommitterFactory` was creating the wrong committer, so if you use `withAutoCommit(false)` it was returning a `KafkaCommitter` instead of the `VoidCommitter`, that should be used in this case. I fixed it in version `1.13.0`. 

@Putr can you please try one more time with the latest version of this package?

**Comment by mateusjunges** [comment]
Closing this for now. Feel free to open another issue or comment here if you are still facing issues with this package. Thanks for reporting!
